src/app/(nextjs_migration)/(authenticated)/user/premium-welcome/page.tsx

```diff
@@ -0,0 +1,17 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use client";
+import { useSession } from "next-auth/react";
+
+export default async function PremiumWelcome() {
+  const { update } = useSession();
+
+  const session = await update();
```

Comment:
@Vinnl hey per [the docs](https://next-auth.js.org/getting-started/client#updating-the-session), we should be able to call `update()` to refresh the JWT from FxA, but I get this:

```
[next-auth][error][CLIENT_FETCH_ERROR] 
https://next-auth.js.org/errors#client_fetch_error Unexpected token o in JSON at position 1 {
  error: {
    message: 'Unexpected token o in JSON at position 1',
    stack: 'SyntaxError: Unexpected token o in JSON at position 1\n' +
      '    at JSON.parse (<anonymous>)\n' +
      '    at Response.json (node:internal/deps/undici/undici:6160:23)\n' +
      '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)',
    name: 'SyntaxError'
  },
  url: 'http://localhost:6060/api/auth/session',
  message: 'Unexpected token o in JSON at position 1'
}
```

Am I using it wrong or is there something wrong with it? The endpoint above seems to return proper JSON.

Reply:
Hmm, I don't get that error. The way it is written now does result in the function getting called every time the component re-renders, which can be often. Usually, you will want to wrap side effects like this into [a `useEffect` call](https://react.dev/learn/synchronizing-with-effects), and since this should run only once when the page first loads, with an empty `dependencies` parameter `[]`. Not sure if that'll solve your issue though, since I can't reproduce.

Reply:
Ah right, that actually did seem to solve the problem. I'll update this to use `useEffect` properly, thanks!

Reply:
OK well that worked, *but* when I test with a new account, it looks like `update` doesn't actually refresh the JWT from FxA unfortunately.

Requiring the user to sign in again works, but I think we should be able to do something like https://next-auth.js.org/v3/tutorials/refresh-token-rotation to do this without having to bother the user.

We could leave this for the front-end work, wdyt @Vinnl?

Reply:
@rhelmer Ah, so as I understand it we need to distinguish two things here. The first are the refresh and access tokens we get from FxA, that verify that the user indeed controls that account. We store those in our database, in the `subscribers` table.

Then there's the JWT that _we_ generate and sign using our secret, in which we state that we have done the FxA verification. When you call `update`, _that_'s the token that gets refreshed.

And if I understand it correctly, you actually want to refresh the profile data we get from FxA. From what I can see, I don't think Next-Auth currently has an officially supported method to do that. However, I think that using the access token that we store in the database, you might be able to replicate [this call](https://github.com/mozilla/blurts-server/blob/fa4f784c789afb1201bffbc696ceef5f5637bf7d/src/app/api/auth/%5B...nextauth%5D/route.ts#L70-L75) to re-fetch the profile data? And if you are able to do so in Next-Auth's [`jwt` callback](https://next-auth.js.org/configuration/callbacks#jwt-callback), you could even persist the new data in the token right away.

Reply:
Oh but yes, we could also defer looking at that until when we implement the UI.